pkg/syncer/syncertest/fake/storage.go

```diff
@@ -68,16 +85,41 @@ type MemoryStorage struct {
 	// The map is indexed by ID.
 	// TODO: The version stored should be the storage version according to the scheme.
 	objects map[core.ID]*unstructured.Unstructured
+	// allowedFieldManagers is the set of field managers allowed to be used.
+	// All other field managers will cause an error.
+	allowedFieldManagers sets.Set[string]
 }
 
 // NewInMemoryStorage constructs a new MemoryStorage
 func NewInMemoryStorage(scheme *runtime.Scheme, watchSupervisor *WatchSupervisor) *MemoryStorage {
 	return &MemoryStorage{
-		scheme:          scheme,
-		watchSupervisor: watchSupervisor,
-		Now:             RealNow,
-		objects:         make(map[core.ID]*unstructured.Unstructured),
+		scheme:               scheme,
+		watchSupervisor:      watchSupervisor,
+		Now:                  RealNow,
+		objects:              make(map[core.ID]*unstructured.Unstructured),
+		allowedFieldManagers: defaultAllowedFieldManagers,
+	}
+}
+
+// SetAllowedFieldManagers allows overriding the default allowed field managers.
+// By default, one of the following must be used:
+// - fake.FieldManager
+// - configsync.FieldManager
+// - reconcilermanager.FieldManager
+//
+// WARNING: fake.DynamicClient does not support passing through field manager
+// due to tech debt in the client-go FakeDynamicClient. So the fake.FieldManager
+// is always used instead.
+// TODO: Pass through FieldManager from Create, Update, & Patch (requires client-go & client-gen changes)
+func (ms *MemoryStorage) SetAllowedFieldManagers(fieldManagers sets.Set[string]) {
```

Comment:
Why do we need to support changing the allowed field managers?

Reply:
I thought it might be useful for downstream code users who want to use the fake clients.